SYCL/ESIMD/spec_const/Inputs/spec_const_common.hpp

```diff
@@ -0,0 +1,89 @@
+//==--------------- spec_const_common.h  - DPC++ ESIMD on-device test -----===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// The test checks that ESIMD kernels support specialization constants for all
+// basic types, particularly a specialization constant can be redifined and
+// correct new value is used after redefinition.
+
+#include "esimd_test_utils.hpp"
+
+#include <CL/sycl.hpp>
+#include <CL/sycl/INTEL/esimd.hpp>
+
+#include <iostream>
+#include <vector>
+
+using namespace cl::sycl;
+
+template <typename AccessorTy>
+ESIMD_INLINE void do_store(AccessorTy acc, int i, spec_const_t val) {
+  using namespace sycl::INTEL::gpu;
+#if STORE == 0
```

Comment:
C++ style is preferred wherever possible (2 more places right below):
```suggestion
  if constexpr (STORE == 0)
```

Reply:
scatter function, that is used in scalar_store, can only be used with types whose size is no larger than 4 bytes:
> llvm/build/install/bin/sycl/CL/sycl/INTEL/esimd/esimd_memory.hpp +338
` typename sycl::detail::enable_if_t<(sizeof(T) <= 4) &&`

Due to the fact that discarded statement is fully checked (according to [cppreference](https://en.cppreference.com/w/cpp/language/if)), compilation of test with type whose size is more than 4 bytes fails:
> llvm/build/install/bin/../include/sycl/CL/sycl/INTEL/esimd/esimd_memory.hpp:342:5: note: candidate template ignored: requirement '(sizeof(double) <= 4) && (1 == 1 || 1 == 8 || 1 == 16) && !std::is_pointer<sycl::accessor<double, 1, sycl::access::mode::write, sycl::access::target::global_buffer, sycl::access::placeholder::false_t, sycl::ONEAPI::accessor_property_list<>>>::value' was not satisfied [with T = double, N = 1, AccessorTy = sycl::accessor<double, 1, sycl::access::mode::write, sycl::access::target::global_buffer, sycl::access::placeholder::false_t, sycl::ONEAPI::accessor_property_list<>>, L1H = sycl::INTEL::gpu::CacheHint::None, L3H = sycl::INTEL::gpu::CacheHint::None]
    scatter(AccessorTy acc, simd<T, N> vals, simd<uint32_t, N> offsets,

Reply:
ok